gobblin-service/src/main/java/org/apache/gobblin/service/modules/db/ServiceDatabaseProviderImpl.java

```diff
@@ -61,25 +60,28 @@ private synchronized void ensureDataSource() {
       return;
     }
 
-    dataSource = new BasicDataSource();
+    dataSource = new HikariDataSource();
 
-    dataSource.setUrl(configuration.getUrl());
+    dataSource.setJdbcUrl(configuration.getUrl());
     dataSource.setUsername(configuration.getUserName());
     dataSource.setPassword(configuration.getPassword());
 
     // MySQL server can timeout a connection so need to validate connections before use
     final String validationQuery = MysqlDataSourceUtils.QUERY_CONNECTION_IS_VALID_AND_NOT_READONLY;
     LOG.info("setting `DataSource` validation query: '" + validationQuery + "'");
-    dataSource.setValidationQuery(validationQuery);
-    dataSource.setValidationQueryTimeout(5);
-
-    // To improve performance, we only check connections on creation, and set a maximum connection lifetime
+    // TODO: revisit following verification of successful connection pool migration:
+    //   If your driver supports JDBC4 we strongly recommend not setting this property. This is for "legacy" drivers
+    //   that do not support the JDBC4 Connection.isValid() API; see:
+    //   https://github.com/brettwooldridge/HikariCP#gear-configuration-knobs-baby
+    dataSource.setConnectionTestQuery(validationQuery);
+    dataSource.setValidationTimeout(5000);
```

Comment:
Nit: Is there a reason we aren't using the `Duration.of` instead of a magic number

Reply:
I guess I didn't consider it since, being a mechanical replacement that originated w/ a magic number.  were you thinking `Duration.ofSeconds(5000).toMillis()` would be more self-documenting?  sure, I could do that.